### PR TITLE
Refactor ETLS/RPC/mux packages

### PR DESCRIPTION
**Pull request by leventeliu**
Redesigned package hierarchies:

- `crypto/etls`: provides crypto connection, doesn't know about `NodeID`
- `noconn`: based on `crypto/etls`, node-oriented connection, depends on a node ID `Resolver`; provides `noconn` Dial/DialEx/Accept methods
- `rpc`: based on `noconn`, defines `Server`/`NodeConnPool` interface/`Caller`/`PersistentCaller`
- `rpc/mux`: multiplexing version of `rpc`; implements a common used version of `Resolver` interface

### CODE REVIEW

File: pow/cpuminer/uint256.go [hunk 1]
```diff
@@ -24,6 +24,10 @@ import (
 	hsp "github.com/CovenantSQL/HashStablePack/marshalhash"
 )
 
+const (
+	Uint256Size = 8 * 4
```
Comment (covenantdb) [inline pow/cpuminer/uint256.go:29]: **[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported const Uint256Size should have comment (or a comment on this block) or be unexported

File: rpc/mux/wrapper.go [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+	return
+}
+
+// WithAcceptConnFunc resets the AcceptConn function of server.
+func (s *Server) WithAcceptConnFunc(f rpc.AcceptConn) *Server {
+	s.Server.WithAcceptConnFunc(f)
+	return s
+}
+
+func NewOneOffMuxConn(conn net.Conn) (net.Conn, error) {
```
Comment (covenantdb) [inline rpc/mux/wrapper.go]: **[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported function NewOneOffMuxConn should have comment or be unexported

File: rpc/client.go [hunk 3]
```diff
@@ -14,205 +14,32 @@
...
-			err = errors.Wrap(err, "open stream failed")
-			return
-		}
-	}
-	client.Conn = muxConn
-	client.Client = rpc.NewClientWithCodec(utils.GetMsgPackClientCodec(muxConn))
-	client.RemoteAddr = conn.RemoteAddr().String()
-
-	return client, nil
+// Client defines the extend method to set client last error.
```
Comment (covenantdb) [inline rpc/client.go]: **[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
comment on exported type LastErrSetter should be of the form "LastErrSetter ..." (with optional leading article)

File: rpc/mux/resolve.go [hunk 4]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright 2018 The CovenantSQL Authors.
+ * Copyright 2019 The CovenantSQL Authors.
```
Comment (laodouya) [inline rpc/mux/resolve.go:2]: 2018-2019

**Review by xq262144 — APPROVED** [review]

**Review by laodouya — APPROVED** [review]

**Pull request merged**